usaspending_api/agencies/README.md

```diff
@@ -0,0 +1,77 @@
+# A Little Background
+
+Whenever an award or subaward is made, there are several federal agencies tied to the award
+including, but not limited to:
+
+* Awarding Agency - The agency that interacted with the recipient to make the award.
+* Funding Agency - The agency that provided the bulk of the funding for the award.
+* Reporting Agency - The agency that reported the award to the system of record (FSRS, DABS, etc).
+
+These agencies may overlap or be missing entirely from the system of record.
+
+Agencies fall into two tiers.
+
+## Toptier Agency
+
+Toptier Agencies are likely what most people think of when they think of government agencies.
+Examples include Department of Defense (DoD), Department of the Treasury (TREAS), Department
+of Health and Human Services (HHS), etc.  There are about 195 of these.
+
+Toptier Agencies are identified by their `cgac_code`.  Unfortunately, that name is a bit of
+a misnomer.  While MOST agencies are identified by their three digit Common Government-Wide
+Accounting Classification (CGAC), there are several identified by their Financial Reporting
+Entity Code (FREC).  This is a point of confusion for many new developers and is on a very
+long TODO list of things to clean up when we have time.
+
+Deciding whether to use CGAC or FREC to identify a toptier agency is controlled by the `is_frec`
+flag in the source agency CSV file.
+
+## Subtier Agency
+
+Subtier Agencies are entities that fall within the purview of a Toptier Agency, for example Office
+of Inspector General within the Department of the Treasury (TREAS) or National Institutes of Health
+(NIH) within the Department of Health and Human Services (HHS).  There are about 1,500 of these.
+
+Subtier agencies are identified by a four digit Subtier code.  The Subtier code list is maintained
+by the product owners of USAspending.
+
+An agency can have many subtiers.  Oftentimes (but not always), one of those subtiers
+represents the agency as a whole.  For example, 020 is Department of the Treasury.  As of
+this writing, Treasury has 45 subtiers.  One of those subtiers, 2000, represents the entirety
+of Treasury.  It is roughly equivalent to 020 but at the subtier level.  2004, on the other
+hand, represents the Office of Inspector General within Treasury and, as such, is not equivalent
+to Treasury as a whole.  `toptier_flag` is used to identify those subtiers that are essentially
+equivalent to their toptier counterpart.  In this example, `toptier_flag` would be True for 2000
+but False for 2004.
+
+## Armed Forces
+
+There are currently four toptier agencies that get rolled up into 097 Department of Defense for
+reporting purposes:
+
+* 017 Department of the Navy
+* 021 Department of the Army
+* 057 Department of the Air Force
+* 096 Corps of Engineers - Civil Works
```

Comment:
Is this one subtier? I thought @sethstoudenmier or @emily-brents  did a thing about that.